MovingBlocks/Terasology
# feat(TypeHandlerLibrary): Add generic Serializer for TypeHandlerLibrary.

### Contains

* Generic Serializer for TypeHandlerLibrary. You can to do this: 
`Optional<SomeClass> obj = serializer.deserializer(TypeInfo.of(SomeClass.class), bytes)`
or
`Optional<SomeClass> obj = serializer.deserializer(TypeInfo.of(SomeClass.class), inputstream)`
Within any code, which use THL.
* mark `ProtobufSerializer` with depricated
* remove `GsonSerializer` (all code migrated to pure `Serializer`
* Migrate `AutoConfigManager` from usage directly `Gson` to usage `Serializer` (AutoConfigSerializer was deleted as unused)
* AutoConfig serialization works via THL now.

This made for using serializer without implicit define concrete format.

### How to test

1. Run tests
2. remove `configs/engine/.....Autoconfig.cfg` (in TS' working directory`)
2. Start TS
3. change AudioConfig's values
4. exit TS
5. check that `configs/engine/....Autoconfig.cfg` have your values.
6. Start TS
7. check that AutoConfig's values equals your saved values.
8. check Record&Replay works( saving recorded events and loading, idk how works R&R. R&R single user of `GsonSerializer`)
